### ISSUE TITLE
Tests cannot be defined in indented blocks

### ISSUE THREAD

**Issue by breuleux** [darrenburns/ward#106]
Defining the following test raises `IndentationError: unexpected indent` during (I assume) collection:

```python
if True:
    @test("Testing indent")
    def _():
        pass
```

I believe the relevant line in the traceback is "ward/rewrite.py", line 96, in rewrite_assertion, `tree = ast.parse(code)`.

This tends to happen when `ast.parse` is called on indented code and can be fixed by using `textwrap.dedent`,  although that would probably mess up column numbers in the resulting AST. An alternative dirty, dirty hack may be to prepend `if True:` to the code if the first non-empty line starts with a space, and then extracting the if body prior to the transform, which would preserve correct column numbers.

**Comment by darrenburns** [comment]
Thanks for the report and for the great feedback earlier :)

**Comment by darrenburns** [comment]
@breuleux - I've got a fix for this in #117 which checks the initial indentation, dedents, parses the tree, then walks the tree and re-indents by the original indentation amount to ensure the column numbers are correct.
